src/ReferenceList.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+	<div class="widgets--list" :class="{'icon-loading': loading }">
+		<div v-for="reference in displayedReferences" :key="reference.openGraphObject.id">
+			<ReferenceWidget :reference="reference" />
+		</div>
+	</div>
+</template>
+<script>
+import ReferenceWidget from './ReferenceWidget.vue'
+import axios from '@nextcloud/axios'
+import { generateOcsUrl } from '@nextcloud/router'
+import { isWidgetRegistered } from './widgets.js'
+export default {
+	name: 'ReferenceList',
+	components: { ReferenceWidget },
+	props: {
+		text: {
+			type: String,
+			default: ''
+		},
+		referenceData: {
+			type: Object,
+			default: null
+		},
+		limit: {
+			type: Number,
+			default: 5
+		}
+	},
+	data() {
+		return {
+			references: null,
+			loading: true
+		}
+	},
+	computed: {
+		firstReference() {
+			return this.references ? Object.values(this.references)[0] : null
+		},
+		displayedReferences() {
+			return this.references ? Object.values(this.references).slice(0, this.limit) : null
+		},
```

Comment:
We can avoid the double usage of `Object.values(this.references)`

```suggestion
	computed: {
	    values () {
			return this.references ? Object.values(this.references) : []
	    },
		firstReference() {
			return this.values[0] ?? null
		},
		displayedReferences() {
			return this.values.slice(0, this.limit)
		},
```